### PR TITLE
Add template country global vars

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      |  Add template country global vars.
| Type?             |improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Related PRs       | N/A
| How to test?      | ```smarty  var_dump($country) ```
| Possible impacts? | N/A

### CODE REVIEW

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -487,6 +487,7 @@ protected function assignGeneralPurposeVariables()
             'cart' => $this->cart_presenter->present($cart),
             'currency' => $this->getTemplateVarCurrency(),
             'customer' => $this->getTemplateVarCustomer(),
+            'country' => $this->getTemplateVarCountry(),
```
Comment (PrestaEdit) [inline classes/controller/FrontController.php]:
```suggestion
            'country' => $this->objectPresenter->present($this->context->country),
```

**Review by PrestaEdit — CHANGES_REQUESTED** [review]
$this->objectPresenter->present($this->context->country)

File: classes/controller/FrontController.php [hunk 2]
```diff
@@ -1552,6 +1553,16 @@ public function getTemplateVarCurrency()
         return $curr;
     }
 
+    public function getTemplateVarCountry()
```
Comment (PrestaEdit) [inline classes/controller/FrontController.php]: No more needed.

**Review by PrestaEdit — APPROVED** [review]

**Review by matks — APPROVED** [review]
Makes sense for me 👍 

Do you have a specific usecase in mind?

**Comment by Hlavtox** [comment]
`Do you have a specific usecase in mind?`

It's good to have everything ready in a template. ;-)

**Comment by matks** [comment]
> `Do you have a specific usecase in mind?`
> 
> It's good to have everything ready in a template. ;-)

They say "don't be afraid of asking questions" so ... I might have missed the joke 😅  but what do you mean?

**Review by FabienPapet — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @PululuK ,

I think I'm testing it wrong. Could you confirm a way to test it?

Here's what I'm doing, in `develop/themes/classic/templates/cms/stores.tpl` I added `{$country}` but the result is not as expected:
Screenshot 2022-10-18 at 11 31 29

Tested with other global vars, same thing. In fact, I know I'm not testing it right 😅 

Waiting for your answer :)
Thanks!

**Review by florine2623 — APPROVED** [review]
Hello again,

Tested again with @Hlavtox 's tips! 😉 

I can access the variable easily just by `{dump($country)}` or adding `{$country.name}`.
Info was added in `MY_SHOP/themes/classic/templates/cms/stores.tpl`
Screenshot 2022-10-18 at 16 00 49
Screenshot 2022-10-18 at 16 04 09




Works like a charm.
It is QA ✅ !

**Comment by matks** [comment]
Thank you @PululuK and @florine2623

**Pull request merged**